### DISCUSSION TITLE
The media must be an image.

### DISCUSSION THREAD

**Discussion by rabichawila** [laravel/nova-issues#3218] [Q&A] [answered]
- Laravel Version: 5.8.*
- Nova Version: 2.0.9
- PHP Version: 7.4.3
- OS: Ubuntu 20.04.1 (LEMP)

### Description:
I get an error saying : `The media must be an image.`, it works fine on localhost(Windows WampServer) though. But not working very well on Production. (Sometimes it works, showtimes it doesn't, i just don't understand what could be wrong.)

Code:

```
    Images::make('Gallery', 'media')
        ->croppable(false)
        ->onlyOnForms()
        ->setFileName(function ($originalFilename, $extension, $model) {
            return time() . '.' . $extension;
        });
    MorphMany::make("Images", "media", Media::class)->exceptOnForms();

```

**Comment by crynobone** [comment, marked as answer]
I honestly don't think Laravel Nova would work well when there's a mixed of relation and attribute sharing the same key.

**Reply by adronitis** [reply]
This issue just popped up for me when running Laravel 5.8.* and Nova 2.12. It was working fine when it was Laravel 5.6.*

**Reply by crynobone** [reply]
Laravel 5.8 and Nova 2 bugfixes support has ended, please upgrade to Nova 3 as soon as possible.
